src/test/ui/hygiene/auxiliary/transparent-basic.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(decl_macro, rustc_attrs)]
+
+#[rustc_transparent_macro]
```

Comment:
Is this attribute only meant for testing?

If so, the rustc-guide should probably get an entry about it

Reply:
Yes, for testing.
It's very inconvenient to only be able to test transparent marks on stage2 with proc macros.
It's going to be replaced with more fine-grained hygiene opt-out (https://github.com/rust-lang/rust/pull/47992) eventually.